src/helpers/channel/allowNotification.ts

```diff
@@ -0,0 +1,22 @@
+import { NOTIF_PERMISSION_STATUS } from './types';
+
+export const handleBrowserNotification = (onSuccess?: () => void) => {
+  if (!('Notification' in window)) {
+    alert('This browser does not support desktop notification');
+  } else if (checkPermission() === NOTIF_PERMISSION_STATUS.PENDING) {
+    Notification.requestPermission().then(() => {
+      onSuccess ? onSuccess() : null;
+    });
+  }
+};
+
+export const checkPermission = () => {
+  if (!('Notification' in window)) {
+    alert('This browser does not support desktop notification');
+  } else if (Notification.permission === 'granted') {
+    return NOTIF_PERMISSION_STATUS.GRANTED;
+  } else if (Notification.permission !== 'denied') {
+    return NOTIF_PERMISSION_STATUS.PENDING;
+  }
+  return NOTIF_PERMISSION_STATUS.DENIED;
+};
```

Comment:
```suggestion
export const checkNotificationPermission = (): NotificationPermissionStatus => {
  if (!('Notification' in window)) {
   return "notValid"
  } else if (Notification.permission === 'granted') {
    return "granted";
  } else if (Notification.permission !== 'denied') {
    return "pending";
  }
  return "denied";
};
```

The alert thing needs to go in a separate function. The notification permission button just needs to return thr relevant permission only.

Feel free to make additional changes.

Reply:
separated the alert function